src/vcpkg/visualstudio.cpp

```diff
@@ -85,14 +146,42 @@ namespace vcpkg::VisualStudio
             return left.version > right.version;
         }
 
-        VisualStudioInstance(Path&& root_path, std::string&& version, const ReleaseType& release_type)
-            : root_path(std::move(root_path)), version(std::move(version)), release_type(release_type)
+        VisualStudioInstance(Path&& root_path,
+                             std::string&& version,
+                             const ReleaseType& release_type,
+                             const bool have_native_desktop = false,
+                             const bool have_core_features = false,
+                             const bool have_msbuild = false,
+                             const bool have_ucrt = false,
+                             const bool have_x86_x64_toolset = false,
+                             const bool have_arm_toolset = false,
+                             const bool have_arm64_toolset = false,
+                             const bool have_uwp_toolset = false)
```

Comment:
having this many bools in the constructor is, imo, a really bad idea. It's extremely confusing. Better to either 1) have an enum for each parameter, or 2) have people initialize the members after construction.

Reply:
Yeah that's ugly, I will improve these code.